lib/pymedphys/_dicom/connect/listen.py

```diff
@@ -0,0 +1,208 @@
+
+        series_uid = dataset.SeriesInstanceUID
+        series_dir = self.storage_directory.joinpath(series_uid)
+        series_dir.mkdir(exist_ok=True)
+        self.association_directory = series_dir
+
+        filename = pathlib.Path(
+            "{0!s}.{1!s}.dcm".format(mode_prefix, dataset.SOPInstanceUID)
+        )
+        filepath = series_dir.joinpath(filename)
+
+        status_ds = pydicom.Dataset()
+        status_ds.Status = 0x0000
+
+        if filepath.exists():
+
+            # If the file already exists, open it up and compare the contents
+            # (converting to JSON string to perform comparison)
+            existing_ds = pydicom.read_file(filepath)
+
+            if not existing_ds.to_json() == dataset.to_json():
+                # If the contents don't match, save conflicting incoming file in an
+                # "orphan" sub-directory.
+
+                # Just give the file a unique name in the orphan directory
+                filename = str(uuid.uuid4())
+                filepath = series_dir.joinpath("orphan", filename)
+                filepath.parent.mkdir(exist_ok=True)
+
+                logging.warning(
+                    "DICOM file exists, storing in orphan directory: %s", filename
+                )
+
+        context = event.context
+        meta = pydicom.Dataset()
+        meta.MediaStorageSOPClassUID = dataset.SOPClassUID
+        meta.MediaStorageSOPInstanceUID = dataset.SOPInstanceUID
+        meta.ImplementationClassUID = pynetdicom.PYNETDICOM_IMPLEMENTATION_UID
+        meta.TransferSyntaxUID = context.transfer_syntax
+
+        # The following is not mandatory, set for convenience
+        meta.ImplementationVersionName = pynetdicom.PYNETDICOM_IMPLEMENTATION_VERSION
+        file_ds = pydicom.FileDataset(
+            filepath, {}, file_meta=meta, preamble=b"\0" * 128
+        )
+        file_ds.update(dataset)
+        file_ds.is_little_endian = context.transfer_syntax.is_little_endian
+        file_ds.is_implicit_VR = context.transfer_syntax.is_implicit_VR
+
+        try:
+            # We use `write_like_original=False` to ensure that a compliant
+            # File Meta Information Header is written
+            file_ds.save_as(filepath, write_like_original=False)
+            status_ds.Status = 0x0000  # Success
+        except IOError:
+            logging.error("Could not write file to specified directory:")
+            logging.error("    %s", filepath)
+            logging.error(
+                "Directory may not exist or you may not have write permission"
+            )
+
+            status_ds.ErrorComment = "SCP internal error - Unable to write file"
+            status_ds.Status = 0xA700  # Failed - Out of Resources - IOError
+
+        return status_ds
+
+
+def listen_cli(args):
+    """Start a DICOM listener from the command line interface
+    """
+
+    # Set log level to debug
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
```

Comment:
Hi @pchlap,

For a follow up PR, we actually already have code built into the pymedphys CLI that allows logging level setting.

For example `pymedphys --debug dicom listen` will cause the log level to be debug, and it puts it in the user control. As such can we remove this loglevel control here?